src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java

```diff
@@ -1490,7 +1490,10 @@ public void testCreateOidcProvider() throws Exception {
             .setDisplayName("DISPLAY_NAME")
             .setEnabled(true)
             .setClientId("CLIENT_ID")
-            .setIssuer("https://oidc.com/issuer");
+            .setClientSecret("CLIENT_SECRET")
+            .setIssuer("https://oidc.com/issuer")
+            .setCodeResponseType(true)
+            .setIdTokenResponseType(true);
```

Comment:
In the Node.js SDK we are adding checks to make sure that exactly one is true, when both code and idToken response types are specified. If we want we can do something similar here too. But it will probably require some additional changes.

In the `AbstractCreateRequest` and `AbstractUpdateRequest` classes:
```java
protected void validate() { }

Map<String, Object> getProperties() {
    this.validate();
    return ImmutableMap.copyOf(properties);
}
```

Now in `OidcProviderConfig.CreateRequest` and `OidcProviderConfig.UpdateRequest`:
```java
private static void validateResponseTypes(Map<String, Object> properties) {
  // Check if clientSecret is set when code = true
  // Check that when both code and idToken are specified, they are not both true or both false
}

@Override
protected void validate() {
  validateResponseTypes(this.properties);
}
```

Take it as a suggestion. I think for the initial release, the way it's implemented now is probably fine.